akka-http-core/src/main/scala/akka/http/impl/engine/client/OutgoingConnectionBlueprint.scala

```diff
@@ -198,7 +198,8 @@ private[http] object OutgoingConnectionBlueprint {
 
       val getNextData = () ⇒ {
         waitingForMethod = false
-        pull(dataInput)
+        if (!isClosed(dataInput)) pull(dataInput)
```

Comment:
I am not sure if this change is correct, but this line sometimes results in trying to call a closed port otherwise.